java/src/main/java/vw/VW.java

```diff
@@ -54,7 +47,44 @@ public static void main(String[] args) {
     public VW(String command) {
         isOpen = true;
         lock = new ReentrantLock();
-        nativePointer = initialize(command);
+        // This allows the user to instead of using the prepackaged JNI layer load their own external JNI layer.
+        // This means that if a user wants to use this code with an OS that is not supported they would follow the following steps
+        // 1.  Download VW
+        // 2.  Build VW for the OS they wish to support
+        // 3.  Call either System.load(xxx) or System.loadLibrary(xxx)
+        // After doing this then the first UnsatisfiedLinkError will never occur.  If a user wishes to use the prepackaed
+        // JNI layer then the first time this object is constructed the UnsatisfiedLinkError will throw and the native libraries
+        // will be loaded and every other time it will not.
+        long currentNativePointer;
+        try {
+            currentNativePointer = initialize(command);
```

Comment:
That is true, but if System.loadLibrary was called then the loadNativeLibrary method will never be called.  In order to make this more clear, I will set loadedNativeLibrary to true in the try block after a successful load.  Also, loadedNativeLibrary is set to true inside of the static lock so it will work correctly.